packages/create-webstone-app/src/index.ts

```diff
@@ -0,0 +1,159 @@
+import execa from "execa";
+import fs from "fs-extra";
+import { Listr } from "listr2";
+import { TaskWrapper } from "listr2/dist/lib/task-wrapper";
+import { DefaultRenderer } from "listr2/dist/renderer/default.renderer";
+import path from "path";
+import prompts from "prompts";
+import readline from "readline";
+
+interface Ctx {
+  appDir: string;
+  webAppDir: string;
+}
+
+interface WebstoneTask extends TaskWrapper<Ctx, typeof DefaultRenderer> {}
+
+const displayWelcome = () => {
+  // https://textfancy.com/ascii-art/
+  console.log(`
+      ▄     ▄ ▄▄▄▄▄▄▄ ▄▄▄▄▄▄▄ ▄▄▄▄▄▄▄ ▄▄▄▄▄▄▄ ▄▄▄▄▄▄▄ ▄▄    ▄ ▄▄▄▄▄▄▄ 
+      █ █ ▄ █ █       █  ▄    █       █       █       █  █  █ █       █
+      █ ██ ██ █    ▄▄▄█ █▄█   █  ▄▄▄▄▄█▄     ▄█   ▄   █   █▄█ █    ▄▄▄█
+      █       █   █▄▄▄█       █ █▄▄▄▄▄  █   █ █  █ █  █       █   █▄▄▄ 
+      █       █    ▄▄▄█  ▄   ██▄▄▄▄▄  █ █   █ █  █▄█  █  ▄    █    ▄▄▄█
+      █   ▄   █   █▄▄▄█ █▄█   █▄▄▄▄▄█ █ █   █ █       █ █ █   █   █▄▄▄ 
+      █▄▄█ █▄▄█▄▄▄▄▄▄▄█▄▄▄▄▄▄▄█▄▄▄▄▄▄▄█ █▄▄▄█ █▄▄▄▄▄▄▄█▄█  █▄▄█▄▄▄▄▄▄▄█
+      
+      `);
+};
+
+const determineAppDirName = async (ctx: Ctx, task: WebstoneTask) => {
+  const appName = process.argv[2];
+  ctx.appDir = appName ? appName.toLowerCase().replace(/\s/g, "-") : ".";
+  task.output = `App directory name: ${ctx.appDir}`;
+  return;
+};
+
+const createAppDir = async (ctx: Ctx, task: WebstoneTask) => {
+  const appDir = ctx.appDir;
+
+  if (fs.existsSync(appDir)) {
+    if (fs.readdirSync(appDir).length > 0) {
+      const response = await task.prompt({
+        type: "confirm",
+        name: "value",
+        message: `The ./${appDir} directory is not empty. Do you want to overwrite it?`,
+        initial: false,
+      });
+
+      if (!response) {
+        process.exit(1);
+      }
+
+      fs.rmSync(appDir, { recursive: true, force: true });
+    }
+  }
+  fs.mkdirSync(appDir, { recursive: true });
+};
+
+const copyTemplate = (ctx: Ctx) => {
+  const templateDir = path.join(__dirname, "template");
+  fs.copySync(templateDir, ctx.appDir);
+};
+
+const initWebApp = async (ctx: Ctx, task: WebstoneTask) => {
+  ctx.webAppDir = `${ctx.appDir}/services/web`;
+
+  fs.removeSync(`${ctx.webAppDir}/.keep`);
+
+  const rl = readline.createInterface({ terminal: true, input: process.stdin });
+  // readline.emitKeypressEvents(process.stdin, rl);
+
+  const svelteSubprocess = execa("npm init svelte@next -y", {
+    cwd: ctx.webAppDir,
+    shell: true,
+    // stdin: rl,
+    // stdout: task.stdout()
```

Comment:
That's where it gets complicated. See discussion I referenced in one of this PR's comments.